test/unit/test_gut.gd

```diff
@@ -221,44 +250,194 @@ func test_delete_all_files_in_a_directory():
 func test_gut_sets_self_on_logger():
 	assert_eq(gr.test_gut.get_logger().get_gut(), gr.test_gut)
 
-func test_simulate_calls_process():
-	var obj = HasProcessMethod.new()
-	gr.test_gut.simulate(obj, 10, .1)
-	gr.test.assert_eq(obj.process_called_count, 10, "_process should have been called 10 times")
-	#using just the numbers didn't work, nor using float.  str worked for some reason and
-	#i'm not sure why.
-	gr.test.assert_eq(str(obj.delta_sum), str(1), "The delta value should have been passed in and summed")
-	assert_pass(2)
-
-func test_simulate_calls_process_on_child_objects():
-	var parent = HasProcessMethod.new()
-	var child = HasProcessMethod.new()
-	parent.add_child(child)
-	gr.test_gut.simulate(parent, 10, .1)
-	gr.test.assert_eq(child.process_called_count, 10, "_process should have been called on the child object too")
-	assert_pass()
+# ------------------------------
+# Simulate tests
+# ------------------------------
 
-func test_simulate_calls_process_on_child_objects_of_child_objects():
+func test_simulate_calls_process_if_object_has_method():
+	var with_method = WithProcess.new()
+	
+	gr.test_gut.simulate(with_method, 5, 0.2)
+	gr.test.assert_eq(with_method.call_count, 5, "_process should have been called 5 times")
+	gr.test.assert_eq(with_method.delta_sum, 1.0, "The delta value should have been passed in and summed")
+	
+	var without_method = WithoutProcess.new()
+
+	gr.test_gut.simulate(without_method, 5, 0.2)
+	gr.test.assert_eq(without_method.call_count, 0, "_process should not have been called")
+
+	assert_pass(3)
+	with_method.queue_free()
+	without_method.queue_free()
+
```

Comment:
Put the original `test_simulate_calls_process_on_child_objects_of_child_objects` back.

```
func test_simulate_calls_process_on_child_objects_of_child_objects():
	var objs = []
	for i in range(5):
		objs.append(HasProcessMethod.new())
		if(i > 0):
			objs[i - 1].add_child(objs[i])
	gr.test_gut.simulate(objs[0], 10, .1)

	for i in range(objs.size()):
		gr.test.assert_eq(objs[i].process_called_count, 10, "_process should have been called on object # " + str(i))

	assert_pass(objs.size())
```